### PR TITLE
feat(datasets): add span to dataset from the trace page

### LINKED ISSUE #3208: [datasets] [ui] add span to dataset from trace page

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3208 
Screenshot 2024-05-23 at 4 13 39 PM

Adds the ability to add a span to a dataset.Has some follow-up UI elements that will be filed separately.

### CODE REVIEW

File: app/src/pages/trace/SpanToDatasetExampleDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+              <Controller
+                control={control}
+                name={"input"}
+                render={({
+                  field: { onChange, onBlur, value },
+                  fieldState: { invalid, error },
+                }) => (
+                  <Card
+                    title="Input"
+                    subTitle="The input to the LLM, prompt, or program"
```
Comment (axiomofjoy) [inline app/src/pages/trace/SpanToDatasetExampleDialog.tsx:215]:
```suggestion
                    subTitle="The input to the LLM, retriever, program, etc."
```

File: app/src/pages/trace/SpanToDatasetExampleDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        });
+      }
+      if (!isJSONObjectString(newExample?.output)) {
+        return setError("output", {
+          message: "Output must be a valid JSON Object",
+        });
+      }
+      if (!isJSONObjectString(newExample?.metadata)) {
+        return setError("metadata", {
+          message: "Metadata must be a valid JSON Object",
```
Comment (axiomofjoy) [inline app/src/pages/trace/SpanToDatasetExampleDialog.tsx:112]:
```suggestion
        return setError("input", {
          message: "Input must be a valid JSON object",
        });
      }
      if (!isJSONObjectString(newExample?.output)) {
        return setError("output", {
          message: "Output must be a valid JSON object",
        });
      }
      if (!isJSONObjectString(newExample?.metadata)) {
        return setError("metadata", {
          message: "Metadata must be a valid JSON object",
```

**Review by axiomofjoy — APPROVED** [review]

**Comment by axiomofjoy** [comment]
Noticing we don't have a close button on this slideover like we do for other slideovers.

Screenshot 2024-05-24 at 1 22 32 PM

**Comment by mikeldking** [comment]
works on my machine? Let me check with you later.
Screenshot 2024-05-24 at 4 57 16 PM

**Pull request merged**